internal/merge/model.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package merge
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Mergeable interface {
+	Get() (map[string]any, error)
+	Walk(map[string]any) error
+	Content() map[string]any
+	Path() string
+}
+
+type DefaultModel struct {
+	content map[string]any
+	path    string
+}
+
+func NewDefaultModel(content map[string]any, path string) *DefaultModel {
```

Comment:
we can probably implement this code using generics, thus improving type safety

Reply:
yup, we should invest some time to check

related task: [rm-2471](https://redmine.sighup-prod.sighup.io/issues/2471)